### PR TITLE
Zap Purchase Path

### PR DESCRIPTION
**Pull request by Padraic-O-Mhuiris**

### CODE REVIEW

File: src/ui/fixedrates/BuyFixedRatesView/BuyFixedRatesZap.tsx [hunk 1]
```diff
@@ -46,27 +61,27 @@ export function BuyFixedRatesZap({
...
-        <TokenAmountInput2
-          showMaxButton
-          placeholder="0.00"
-          inputGroupStyle={{
-            height: "72px",
-            width: "100%",
-            fontSize: "1.125rem",
+        <div
+          style={{
+            background: isDarkMode ? "var(--bp3-dark-bg-color)" : "",
```
Comment (sentilesdal) [inline src/ui/fixedrates/BuyFixedRatesView/BuyFixedRatesZap.tsx:66]: use tailwind for this instead

File: src/ui/fixedrates/BuyFixedRatesView/BuyFixedRatesZap.tsx [hunk 2]
```diff
@@ -46,27 +61,27 @@ export function BuyFixedRatesZap({
...
-          maxAmount={inputAssetBalanceOf}
-          tokenDecimals={inputToken.decimals}
-          onValueChange={onInputChange}
-        />
+          className={classNames(tw("flex", "rounded", "mb-2"))}
+        >
+          <TokenAmountInput2
+            showMaxButton
+            placeholder="0.00"
+            inputGroupStyle={{
```
Comment (sentilesdal) [inline src/ui/fixedrates/BuyFixedRatesView/BuyFixedRatesZap.tsx:73]: can we move away from using styles?  we should stick to tailwind as much as possible unless we have to override third party components

File: src/ui/fixedrates/BuyFixedRatesView/BuyFixedRatesZap.tsx [hunk 3]
```diff
@@ -81,7 +96,36 @@ export function BuyFixedRatesZap({
...
+            )}
+          >
+            {marketRateLabel}
+          </span>
+        )}
       </div>
+      <div className={tw("flex", "flex-col", "space-y-3")}>
+        <span className={tw("text-base", "text-left")}>{t`Review Order`}</span>
+
+        {/* <FixedRatePreviewCallout
```
Comment (sentilesdal) [inline src/ui/fixedrates/BuyFixedRatesView/BuyFixedRatesZap.tsx]: remove dead code?

File: src/elf/curve/tokens.ts [hunk 4]
```diff
@@ -5,17 +5,28 @@ import {
...
+  readonly CurvePoolToken: unique symbol;
+};
 
 export function isCurveLpToken(
   tokenInfo: TokenInfo
 ): tokenInfo is CurveLpTokenInfo {
   return !!tokenInfo?.tags?.includes(TokenTag.CURVE);
 }
 
+// Returns the constituent tokenInfos of the tokens which are members of curve
```
Comment (sentilesdal) [inline src/elf/curve/tokens.ts]: nit pick, but try to use doc blocks for function comments, lotta examples around the code base

File: src/elf/zaps/zapPurchase/path.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+interface ZapPurchasePathDoubleStep {
+  kind: ZapPurchasePathKind.DoubleStep;
+  principalToken: PrincipalTokenInfo;
+  baseToken: CurveLpTokenInfo;
+  metaToken: CurveLpTokenInfo;
+  curvePoolToken: TokenInfo;
+}
+
+//
+// ZapPurchasePath is used to signify the relationship between the
```
Comment (sentilesdal) [inline src/elf/zaps/zapPurchase/path.ts]: omg, thanks for the large comment <3

Comment (sentilesdal) [reply src/elf/zaps/zapPurchase/path.ts]: oh also, try to use doc blocks :P

File: src/elf/curve/tokens.ts [hunk 6]
```diff
@@ -27,6 +27,25 @@ export function getCurvePoolTokensByCurveLpToken(
 // e.g The MIM-3LP principal token uses an underying token of the same name, its
 // corresponding pool has token members, MIM and 3CRV. As 3CRV is also a curve lp
 // token, we also find the members of its pool, DAI, USDC, and USDT
+//
+// Visually the hierarchy follows three cases:
```
Comment (sentilesdal) [inline src/elf/curve/tokens.ts]: nice

File: src/ui/fixedrates/BuyFixedRatesView/BuyFixedRatesZap.tsx (hunk 1 above)
Comment (Padraic-O-Mhuiris) [reply src/ui/fixedrates/BuyFixedRatesView/BuyFixedRatesZap.tsx:66]: This was copied from original BuyFixedRateView. I can change again in a UI cleanup PR once closer to finishing?

File: src/ui/fixedrates/BuyFixedRatesView/BuyFixedRatesZap.tsx (hunk 2 above)
Comment (Padraic-O-Mhuiris) [reply src/ui/fixedrates/BuyFixedRatesView/BuyFixedRatesZap.tsx:73]: Same reply as above

File: src/elf/curve/tokens.ts (hunk 4 above)
Comment (Padraic-O-Mhuiris) [reply src/elf/curve/tokens.ts]: Will do doc blocks

File: src/ui/fixedrates/BuyFixedRatesView/BuyFixedRatesZap.tsx [hunk 7]
```diff
@@ -81,7 +96,36 @@ export function BuyFixedRatesZap({
...
+            principalTokensOut={principalTokensOut}
+            baseAssetIn={baseAssetInputValue}
+            baseAssetDecimals={baseAssetDecimals}
+            /> */}
+        <div>Fixed Preview Callout</div>
```
Comment (DannyDelott) [inline src/ui/fixedrates/BuyFixedRatesView/BuyFixedRatesZap.tsx:112]: missing `t` tag

**Review by DannyDelott — APPROVED** [review]
Looks good w/ deadcode removed-- unless this is fixed in a subsequent PR already, that is!

**Pull request merged**